src/main/java/org/example/daos/JobRoleDao.java

```diff
@@ -105,4 +106,449 @@ public JobRoleDetailResponse getJobRoleByID(final int id)
         return null;
     }
 
+    /**
+     * DAO method to order job role names in ascending order from the database.
+     *
+     */
+    public List<JobRoleResponse> jobNameAscending()
```

Comment:
This can be a lot simpler, as right now we've a load of code and a lot of it's repeated. Keep your getJobRoles (or however it is named) function and pass in the two parameters of heading, direction and if you have those fields - simply edit your sql query to contain them.